File: code/mathematical_algorithms/src/average_stream_numbers/average_stream_numbers.rb
# Part of Cosmos by OpenGenus Foundation

def average_stream_number(input_array)
  avg = 0
  resultant_array = []
  input_array.each_with_index do |element, i|
    avg = average(element, avg, (i + 1))
    puts avg
    resultant_array << avg
  end
  resultant_array
end

def average(new_element, prev_avg, index)
  ((prev_avg * (index - 1) + new_element).to_f / index)
end

# Dynamic array input
puts 'Enter the size of array'
size_of_array = gets.chomp.to_i
input_array = []
size_of_array.times.each do
  input_array << gets.chomp.to_i
end

average_stream_number(input_array)

File: code/mathematical_algorithms/src/average_stream_numbers/average_stream_numbers.py
# Part of Cosmos by OpenGenus Foundation
def newAvg(prevAvg, newN, newX):
    return (prevAvg * (newN - 1) + newX) / newN


def main():
    L = [1, 9, 20, 13, 45]
    avg = 0
    for i in range(len(L)):
        avg = newAvg(avg, (i + 1), L[i])
        print(avg)


main()
